plugins/thunderbolt/fu-thunderbolt-plugin.c

```diff
@@ -31,11 +39,19 @@ static gboolean
 fu_thunderbolt_plugin_device_created(FuPlugin *plugin, FuDevice *dev, GError **error)
 {
 	FuContext *ctx = fu_plugin_get_context(plugin);
+	FuThunderboltPlugin *self = FU_THUNDERBOLT_PLUGIN(plugin);
+
 	fu_plugin_add_rule(plugin,
 			   FU_PLUGIN_RULE_INHIBITS_IDLE,
 			   "thunderbolt requires device wakeup");
 	if (fu_context_has_hwid_flag(ctx, "retimer-offline-mode"))
 		fu_device_add_private_flag(dev, FU_THUNDERBOLT_DEVICE_FLAG_FORCE_ENUMERATION);
+
+	/* by default usb4_port1 is used, but this needs to be configurable with a DMI match */
```

Comment:
dmi match?  that feels incredibly hacky.  Why does it change for the DMI of a given machine?  Why can't it be discovered?

Reply:
Agreed; I've added a path search in https://github.com/fwupd/fwupd/pull/7206